### PR TITLE
refactor: custom network component

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

This PR refactors the CustomNetwork component to include it in [#7169 feat(ramp): add NetworkSwitcher view](https://github.com/MetaMask/metamask-mobile/pull/7169) 

**Changes**
**Add**

- Optional prop `onNetworkSwitch` that overrides default navigation behavior
- Optional prop `showAddedNetworks` to show currently added network, changing the action from "Add" to "Switch" network
- Optional prop `customNetworksList` that overrides the current `PopularList` variable being read inside the component
- `"Switch"` string

**Change** 
- Change `ImageIcon` component to a `AvatarNetwork` component that will take `imageSource` or `{ uri: imageUrl }`, defaulting to the circled network initial letter in case none is present
- Hide the warning icon if `toggleWarningModal` is not passed as prop
- Refactor `Network` interface to add `rpcPrefs.imageUrl` as optional property
- Remove `CustomNetworkProps` interface extension of `Networks` 

**QA** 
This PR must not change current only reference of this component https://github.com/MetaMask/metamask-mobile/blob/5d964f6df233a49ef5da6f581a753b1414305db0/app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js#L1092-L1102 

**Screenshots/Recordings**

~~ _If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

fixes #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx [hunk 1]
```diff
@@ -83,7 +100,11 @@ const CustomNetwork = ({
                 onPress={toggleWarningModal}
               />
             ) : null}
-            <CustomText link>{strings('networks.add')}</CustomText>
+            <CustomText link>
+              {networkConfiguration.isAdded
+                ? strings('networks.switch')
+                : strings('networks.add')}
+            </CustomText>
```
Comment (georgewrmarshall) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx:107]: Should we be using `Text` or `ButtonLink` here from the component library? cc @brianacnguyen

Comment (wachunei) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx:107]: `<CustomText link>` is not part of the changes as I don't want to introduce new imports to the component 😬

**Review by chrisleewilcox — DISMISSED** [review]
@wachunei please run the smoketests against this branch on bitrise.  Let us know if you need assistance on doing that.

**Comment by wachunei** [comment]
Bitrise pipelines started here https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/f051934b-663a-49ac-bc06-0dc0839cb9af cc @chrisleewilcox

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx:107]: @georgewrmarshall I will resolve this conversation. 

For more context: we had a sync with @georgewrmarshall & @brianacnguyen, we will work together as Design System <> Ramp teams for the next refactor of Ramps to highly adopt components from the component library in the future, estimated time is next Q1.

Comment (georgewrmarshall) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx:107]: Thanks @wachunei 👍 Don't want to hold this PR up but if we can use more component library components in future that would be great. @brianacnguyen will do a knowledge share once storybook is fixed

**Comment by wachunei** [comment]
@chrisleewilcox should I re run smoke tests?

**Comment by wachunei** [comment]
Passing smoke tests: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/c9f35e14-4f47-4a38-9c0a-f677162d58ca

**Review by tommasini — APPROVED** [review]
Awesome! LGTM!

**Pull request merged**